rotkehlchen/api/rest.py

```diff
@@ -3163,56 +3166,51 @@ def get_current_assets_price(
         }
         return _wrap_in_ok_result(process_result(result))
 
-    def query_location_asset_mappings(self, filter_query: LocationAssetMappingsFilterQuery) -> Response:  # noqa: E501
-        """Query the location asset mappings using the provided filter_query
+    def query_asset_mappings_by_type(
+            self,
+            mapping_type: Literal['location', 'counterparty'],
+            filter_query: LocationAssetMappingsFilterQuery | CounterpartyAssetMappingsFilterQuery,
+    ) -> Response:
+        """Query the location/counterparty asset mappings using the provided filter_query
         and return them in a paginated format"""
-        mappings, mappings_found, mappings_total = GlobalDBHandler.query_location_asset_mappings(
-            filter_query=filter_query,
-        )
+        if mapping_type == 'location':
```

Comment:
Why do you still need this if? Can't; you add the stuff such ad dict keys, query columns and callback as arguments  in the function? Or if not just make two functions.